crowbar_framework/app/models/node_object.rb

```diff
@@ -256,6 +256,32 @@ def intended_role=(value)
     @node["crowbar_wall"]["intended_role"] = value
   end
 
+  def disk_roles=(value)
+    @node["crowbar_wall"] ||= {}
+    @node["crowbar_wall"]["disk_roles"] = value
+  end
+
+  def disk_roles
+    disk_roles = crowbar_wall["disk_roles"] || {}
+
+    # remove devices that were claimed after they role was set
+    disk_roles.each do |d, _role|
+      crowbar_wall["disk_roles"].delete d if crowbar_wall[:claimed_disks].key? d
+    end
+
+    # Check for current unclaimed disks, and add unclaimed disk to
+    # disk_roles hash if it is not there yet.
+    # (New disk could have been added to the node)
+    unclaimed_physical_drives.each do |d, _data|
+      unique_name = unique_device_for(d)
+      unless disk_roles[unique_name]
+        crowbar_wall["disk_roles"] ||= {}
+        crowbar_wall["disk_roles"][unique_name] = ""
+      end
+    end
+    crowbar_wall["disk_roles"] || {}
```

Comment:
The 'side-effecting' setter method is necessary, because the disks can be out of sync when we need the list IIUC. But maybe it'd read a bit better if the 'sync' operations would be in separate methods, e.g.:

```ruby
def disk_roles
   drop_claimed_disks
   add_unclaimed_disks

   crowbar_wall["disk_roles"]
end
```